src/rust/src/buf.rs

```diff
@@ -19,6 +21,13 @@ fn _extract_buffer_length<'p>(
     mutable: bool,
 ) -> pyo3::PyResult<(pyo3::Bound<'p, pyo3::PyAny>, usize)> {
     let py = pyobj.py();
+    if pyobj.is_instance_of::<pyo3::types::PyString>() {
+        return Err(PyErr::new::<PyTypeError, _>(
```

Comment:
You can use `PyTypeError::new_err("...")` instead.

We generally write out the full path `pyo3::exception::PyTypeError::new_err()`.